### ISSUE TITLE
bug in parsing array param in body

### ISSUE THREAD

**Issue by sdghchj** [swaggo/swag#565]
I have a comment as:
```
// @Param body body []integer true "array integer"
```
However it was parsed as follow in my docs.go:
```
{
    "description": "array integer",
    "name": "body",
    "in": "body",
    "required": true,
    "schema": {
                "type": "integer",   //here I think it shoud be "array"
                "items": {
                     "type": ["integer"]
                 }
    }
}
```

Schema type was parsed as integer, so it was displayed as follows :
![image](https://user-images.githubusercontent.com/20944466/69299031-7fd4ff00-0c4a-11ea-8e84-660f9188e280.png)
![image](https://user-images.githubusercontent.com/20944466/69299063-90857500-0c4a-11ea-8764-734667962964.png)

But I want it to be displayed as array. After I edited my docs.go as :
```
{
    "description": "array integer",
    "name": "body",
    "in": "body",
    "required": true,
    "schema": {
                "type": "array", 
                "items": {
                     "type": "integer"
                 }
    }
}
```
then displayed as what I expected:
![image](https://user-images.githubusercontent.com/20944466/69299219-fe31a100-0c4a-11ea-86c5-ec98187a3404.png)
![image](https://user-images.githubusercontent.com/20944466/69299232-05f14580-0c4b-11ea-88ff-05a826e86dd0.png)

I found code [here](https://github.com/swaggo/swag/blob/v1.6.3/operation.go#L156) param.Schema.Type is default assigned by refType,which I think shoud be by objectType.
```
param := createParameter(paramType, description, name, refType, required)
```

and from [here](https://github.com/swaggo/swag/blob/v1.6.3/operation.go#L183), param.Schema.Type  have still not been reassigned . I think it's a bug.

**Comment by ubogdan** [comment]
Since integer is not a valid go primitive we can guess integer is a user-defined type. Try 
```
// @Param body body []int true "array integer"
```

**Comment by sdghchj** [comment]
@ubogdan 
Read code  [here](https://github.com/swaggo/swag/blob/v1.6.3/operation.go#L190) and [swagger primitive](https://github.com/swaggo/swag/blob/ddfc535c4657167b5127ffbdcfb8406deafee1ef/schema.go#L14) please.

int is considered to be user-defined type, not integer

**Comment by ubogdan** [comment]
OK. Thanks.

**Comment by sdghchj** [comment]
I modified [code](https://github.com/swaggo/swag/blob/v1.6.3/operation.go#L179)
```
case "body":
		switch objectType {
		case "primitive":
			param.Schema.Type = spec.StringOrArray{refType}
		case "array":
			param.Schema.Type = spec.StringOrArray{objectType} // add this line
			param.Schema.Items = &spec.SchemaOrArray{
				Schema: &spec.Schema{
					SchemaProps: spec.SchemaProps{},
				},
			}
			// Array of Primitive or Object
			if IsPrimitiveType(refType) {
                                //add following two lines
				param.SimpleSchema.Type = objectType
				param.SimpleSchema.Items = &spec.Items{SimpleSchema: spec.SimpleSchema{Type: refType}}
				param.Schema.Items.Schema.Type = spec.StringOrArray{refType}
			} else {
				if err := operation.registerSchemaType(refType, astFile); err != nil {
					return err
				}
				param.Schema.Items.Schema.Ref = spec.Ref{Ref: jsonreference.MustCreateRef("#/definitions/" + refType)}
			}
```

then displayed:

![image](https://user-images.githubusercontent.com/20944466/69392384-d442b200-0d10-11ea-8dce-b2ff3a1cba3f.png)

This is perfect for me.

**Comment by Canecat** [comment]
@sdghchj Could you post the pull request for this issue? :)

**Comment by easonlin404** [comment]
fix via #579. please reopen it if there are still any issues.
